BraveRewardsUI/SKUs/SKUPurchaseViewController.swift

```diff
@@ -0,0 +1,139 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import BraveRewards
+import BraveShared
+
+public class SKUPurchaseViewController: UIViewController, UIViewControllerTransitioningDelegate {
+  
+  private let rewards: BraveRewards
+  private let publisher: PublisherInfo
+  private let request: PaymentRequest
+  private let ledgerObserver: LedgerObserver
+  private var openBraveTermsOfSale: () -> Void
+  private let responseHandler: (_ response: PaymentRequestResponse) -> Void
+  
+  public init(
+    rewards: BraveRewards,
+    publisher: PublisherInfo,
+    request: PaymentRequest,
+    responseHandler: @escaping (_ response: PaymentRequestResponse) -> Void,
+    openBraveTermsOfSale: @escaping () -> Void
+  ) {
+    self.rewards = rewards
+    self.publisher = publisher
+    self.request = request
+    self.responseHandler = responseHandler
+    self.openBraveTermsOfSale = openBraveTermsOfSale
+    self.ledgerObserver = LedgerObserver(ledger: rewards.ledger)
+    
+    super.init(nibName: nil, bundle: nil)
+    
+    modalPresentationStyle = .overCurrentContext
+    if #available(iOS 13.0, *) {
+      isModalInPresentation = true
+    }
+    transitioningDelegate = self
+    
+    self.rewards.ledger.add(self.ledgerObserver)
+    setupLedgerObserver()
+    
+    rewards.ledger.fetchBalance(nil)
+  }
+  
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+  
+  private var purchaseView: SKUPurchaseView {
+    return view as! SKUPurchaseView // swiftlint:disable:this force_cast
+  }
+  
+  public override func loadView() {
+    view = SKUPurchaseView()
+  }
+  
+  public override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    title = Strings.SKUPurchaseTitle
+    
+    purchaseView.detailView.itemDetailValueLabel.text = request.details.displayItems.map { $0.label }.joined(separator: ", ")
+    purchaseView.detailView.orderAmountLabels.amountLabel.text = request.details.total.amount.value
```

Comment:
Some checks will be done on the JS side as well: https://github.com/brave/brave-ios/issues/2420 - We should probably do the checks in the client code as well. Let me know if you'd prefer a different PR for error handling.

Reply:
Don't we already do that check here? https://github.com/brave/brave-ios/pull/2414/files#diff-4dc70f536f408cb9c86ed24f1c0394eeR67

Reply:
That's just one of the check. We'll need to handle errors where:

1. Total amount doesn't match the individual items. 
2. Amount is not an integer

Also, since we're only showing value in BAT. We need to check if the request currency is BAT and then show the converted value if its not BAT. Let me know if you prefer a separate PR for that.

Reply:
I still don't follow since I wasn't aware any of that could happen. If its not BAT we shouldn't be processing it at all on iOS.

What's supposed to happen if these checks fail?

Reply:
Will handle this as part of #2420